docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+######################
+User Interface Design
+######################
+
+This document outlines the design principles and guidelines for the user
+interface (UI) of NWChemEx. Our main goal is to provide a user-friendly and
+intuitive interface that enables users to carry out a variety of quantum
+chemistry calculations seamlessly. 
+
+
+Design Principles
+=================
+
+Performant
+-----------
+
+Performance is critical for NWChemEx with kernels supporting optimized MPI
+parallelization and GPU offloading. The UI is designed to uphold this
+performance standard and to provide the user with critical control elements to
+achieve high performance. This requires keeping the overhead of UI as minimal as
+possible by avoiding unnecessary copies or data movements and enabling user to
+access and modify the MPI communicator and GPU offloading as needed.
+ 
+Pythonic
+---------
+
+While NWChemEx is mainly written in C++, the UI is designed for and implemented
+in Python, which is arguably one of the most intuitive and widely-used
+programming languages in scientific computing. Its extensive ecosystem and
+user-friendly syntax allow for quicker prototyping and easier integration with
+external tools, making it an ideal choice for this interface. Adhering to the
+Pythonic style of programming is preferred due to its emphasis on code
+readability and simplicity, as described in the Zen of Python (PEP 20)
+<https://peps.python.org/pep-0020/>_.
```

Comment:
I think I'd push back a little. I think this is fine here because it does inform the UI design, and it isn't too repetitive. Maybe just add that link that into the text here?

Reply:
It is possible to have Python API and have a DSL-based UI like Psithon, so I think the discussion should stay. I added a link to the API [discussion](https://nwchemex-project.github.io/.github/design/overall_justifications/why_python.html). See 6230d5cfb57d167f90e77e4f71ddacda666046b2.